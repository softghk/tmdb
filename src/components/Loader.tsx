import React from 'react'

export const Loader = () => {
  return <div className='text-center my-10'>Loading ...</div>
}
