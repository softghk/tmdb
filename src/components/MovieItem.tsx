import {MovieItemType} from '@types'
import Image from 'next/image'
import React from 'react'
import {getImage} from '@utils'
import StarIcon from '@mui/icons-material/Star'
import {Button} from '@mui/material'

export interface MovieItemProps {
  item: MovieItemType
  isFavorite: boolean
  handleFavorite: (id: number) => void
}

const MovieItem = ({item, isFavorite, handleFavorite}: MovieItemProps) => {
  if (item.poster_path) {
    return (
      <a
        href={`${process.env.NEXT_PUBLIC_BASE}movie/${item.id}`}
        target='_blank'
        className={`border-zinc-800 p-2 border hover:scale-105 hover:shadow-lg transition-all rounded-lg overflow-hidden ${
          isFavorite ? 'bg-blue-200' : 'bg-white'
        }`}
      >
        <Image
          loader={() => 'Loading ...'}
          src={getImage(item.poster_path)}
          alt={item.title}
          width={300}
          className='w-full rounded-lg'
          height={300}
        />
        <div className='p-1'>
          <div className='flex justify-between text-xs text-neutral-400'>
            <span>{item.vote_average > 0 && <>IMDB: {item.vote_average}</>}</span>
            <span>{item.release_date?.slice(0, 4)}</span>
          </div>
          <div className='flex justify-between items-start'>
            <h5>{item.title}</h5>
            {/* Favorite Button */}
            <Button
              onClick={(e) => {
                e.preventDefault()
                handleFavorite(item.id)
              }}
              size='small'
              color='warning'
              data-testid="favorite"
            >
              <StarIcon color='warning' />
            </Button>
          </div>
        </div>
      </a>
    )
  }

  return null
}

export default MovieItem
