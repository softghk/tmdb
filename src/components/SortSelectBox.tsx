import {MenuItem, Select} from '@mui/material'
import {SORT_TYPE} from '@services'
import React from 'react'

interface SortSelectBoxProps {
  value: SORT_TYPE
  onChange: (e: SORT_TYPE) => void
}

export const SortSelectBox = ({value, onChange}: SortSelectBoxProps) => {
  return (
    <Select
      value={value}
      defaultValue={value}
      onChange={(e) => onChange(e.target.value as SORT_TYPE)}
      variant='outlined'
      data-testid='sort-select' // Add this data-testid
    >
      <MenuItem value={SORT_TYPE.ASC}>ASC</MenuItem>
      <MenuItem value={SORT_TYPE.DESC}>DESC</MenuItem>
    </Select>
  )
}
