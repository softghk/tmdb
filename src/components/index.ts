export * from './MovieItem'
export * from './Loader'
export * from './SortSelectBox'
