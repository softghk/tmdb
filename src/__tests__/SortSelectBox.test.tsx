import React from 'react'
import {fireEvent, render} from '@testing-library/react'
import {SortSelectBox} from '../components/SortSelectBox'
import {SORT_TYPE} from '@services'

describe('SortSelectBox component renders and handles change', () => {
  const handleChange = jest.fn()

  const ScreenSelect = <SortSelectBox value={SORT_TYPE.DESC} onChange={handleChange} />
  it('renders SortSelect unchanged', () => {
    const {container} = render(ScreenSelect)
    expect(container).toMatchSnapshot()
  })

  it('displays the initial value correctly', () => {
    const {getByTestId} = render(ScreenSelect)
    const selectElement = getByTestId('sort-select').querySelector(
      'input.MuiSelect-nativeInput'
    ) as HTMLInputElement
    expect(selectElement.value).toEqual(SORT_TYPE.DESC)
  })

  it('calls the onChange handler when a different option is selected', () => {
    const {getByTestId} = render(ScreenSelect)
    const selectElement = getByTestId('sort-select').querySelector(
      'input.MuiSelect-nativeInput'
    ) as HTMLInputElement

    fireEvent.change(selectElement, {
      target: {value: SORT_TYPE.ASC},
    })

    expect(handleChange).toHaveBeenCalledWith(SORT_TYPE.ASC)
  })
  it('displays the updated value correctly after a change', () => {
    const {getByTestId} = render(ScreenSelect)
    const selectElement = getByTestId('sort-select').querySelector(
      'input.MuiSelect-nativeInput'
    ) as HTMLInputElement

    fireEvent.change(selectElement, {
      target: {value: SORT_TYPE.DESC},
    })

    expect(selectElement.value).toEqual(SORT_TYPE.DESC)
  })
})
