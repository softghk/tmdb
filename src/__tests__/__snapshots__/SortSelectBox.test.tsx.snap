// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortSelectBox component renders and handles change renders SortSelect unchanged 1`] = `
<div>
  <div
    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
    data-testid="sort-select"
  >
    <div
      aria-controls=":r0:"
      aria-expanded="false"
      aria-haspopup="listbox"
      class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      role="combobox"
      tabindex="0"
    >
      DESC
    </div>
    <input
      aria-hidden="true"
      aria-invalid="false"
      class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      tabindex="-1"
      value="desc"
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden="true"
      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        class="css-ihdtdm"
      >
        <span
          class="notranslate"
        >
          â€‹
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
